src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -192,12 +192,30 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
                 // Set the default pipelineId to 0, which is an invalid build id and it has to be reassigned to a valid build id.
                 int pipelineId = 0;
 
-                bool pipelineTriggeringBool = false;
+                bool pipelineTriggeringBool;
                 if (bool.TryParse(pipelineTriggering, out pipelineTriggeringBool) && pipelineTriggeringBool)
                 {
-                    string triggeringPipeline = context.Variables.GetValueOrDefault("build.triggeredBy.buildId")?.Value;
+                    string hostType = context.Variables.GetValueOrDefault("system.hostType").Value;
```

Comment:
```suggestion
                    string hostType = context.Variables.GetValueOrDefault("system.hostType")?.Value;
```

Reply:
If that value is null, we need to it to fail, since something is very wrong with the env, hence no ?